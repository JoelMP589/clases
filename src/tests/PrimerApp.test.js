import { render } from "@testing-library/react";
import { shallow } from "enzyme/build";
import PrimeraApp from "../PrimeraApp";

describe('pruebas en nuestro componente <PrimeraApp />', () => {
     test('debe de mostrar el mensaje "Hola soy un saludo"', () => {
        const saludo = "Hola soy un saludo";
        const { getByText } = render(<PrimeraApp saludo={saludo} />);
        expect(getByText(saludo)).toBeInTheDocument();
    }); 
    test('debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = "Hola soy un saludo!!!";
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    });

    //comproar elementos dentro de un componente 
    test('debe de mostrar el subtitulo enviado por argumentos', () => {
        const saludo = "Hola soy un saludo!!!";
        const wrapper = shallow(<PrimeraApp
            subtitulo="este es el subtitulo"
            saludo={saludo} />);
        expect(wrapper.find('p').text()).toBe('este es el subtitulo');
    })

});
