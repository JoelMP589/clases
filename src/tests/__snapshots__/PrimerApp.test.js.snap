// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en nuestro componente <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy un saludo!!!
  </h1>
  <p>
    propiedad por defecto
  </p>
</Fragment>
`;
